surmise/calibration.py

```diff
@@ -259,11 +265,37 @@ def predict(self, x=None, args=None):
         if 'predict' in dir(self.method):
             self.method.predict(info, self.info, self.emu, x, args)
         else:
-            emupred = self.emu.predict(x, self.theta.rnd(1000))
+            nsamp = 1000
+            emupred = self.emu.predict(x, self.theta.rnd(nsamp))
             info['mean'] = np.mean(emupred.mean(), 1)
             info['var'] = np.var(emupred.mean(), 1)
             info['rnd'] = (emupred.mean()).T
-        return prediction(info, self)
+
+        predobj = prediction(info, self)
+        predobj.empirical_coverage()
+        return predobj
+
+    @staticmethod
+    def cast_f64_dtype(x):
+        return np.array(x, dtype=np.float64)
+
+    def save(self, filename):
+        """
+        Simple serialization and save function for calibrator object.
+
+        :Example:
+        >>> import dill
+        >>> cal = calibrator(...)
+        >>> # save calibrator object
+        >>> cal.save('cal_example.pkl')
+        >>> # load calibrator object
+        >>> with open('cal_example.pkl', 'rb') as f:
+        >>>     loaded_cal = dill.load(f)
+        """
+        with open(filename, 'wb') as f:
+            dill.dump(self, f)
+        f.close()

```

Comment:
My understanding is that `f` should be closed automatically when Python exits your `with` context manager block.

I'm curious to know why you add the `return` statement. Are you secretly an old-school Fortran programmer?!